### PR TITLE
Add first application

### PR DESCRIPTION
**Pull request by RoboticsYY**
This PR intends to initialize the directory structure of the project and add the first application example. 

**Directory Structure.** The code and tutorial of an application are distributed in `examples` and `tutorials` sub-folders respectively. Under `examples`, each application has it own folder which consists of its `moveit_config` package and `code` package. 

**First application.** The first example demonstrates how to integrate [OpenVINO accelerated GPD](https://github.com/sharronliu/gpd/tree/libgpd) and MoveIt pick-place pipeline. The [tutorials](https://roboticsyy.github.io/moveit_example_apps/index.html) of moveit_example_apps and its first application are also included in this PR to facilitate result replication. It is also possible to see the same execution result on a UR5 robot from the [video](https://www.youtube.com/embed/b4EPvHdidOA?rel=0).

**Docker Container and CI Test** are in development.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,95 @@
...
+
+| Application Name | Documentation          | Docker |
+| ---------------- | -----------------------| ------ |
+| Random Pick      | [random_pick_tutorial] | WIP    |
+
+[random_pick_tutorial]: https://roboticsyy.github.io/moveit_example_apps/doc/random_pick/random_pick.html
+
+## Continuous Integration
+
+Work in progress.
```
Comment (davetcoleman) [inline README.md:15]: Can we do this now, using moveit_ci? Its not too difficult to setup, you can probably just copy the configuration from https://github.com/ros-planning/moveit_grasps/blob/melodic-devel/.travis.yml

File: README.md [hunk 2]
```diff
@@ -1,2 +1,95 @@
...
+
+There are many diverse application examples of what you can use MoveIt for. Your contribution is encouraged, no matter the application is developed for some usages, competitions or research topics, or to demonstrate the newly developed MoveIt feature.
+
+If you are interested in adding a new MoveIt application in this project, please consider following the below instructions.
+
+### Directory Structure
+
+For each example application, it should contain two parts: example codes and tutorials.
+
+* The example codes of each application should live in its own subdirectory in the `./examples/<app_name>` direcotry.
```
Comment (davetcoleman) [inline README.md:27]:
```suggestion
* The example codes of each application should live in its own subdirectory in the `./examples/<app_name>` directory.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,95 @@
...
+* Under `./examples/<app_name>` direcotry, there should be two packages: `<app_name>` and `<app_moveit_config_package>`.
```
Comment (davetcoleman) [inline README.md:28]: directory

File: README.md [hunk 4]
```diff
@@ -1,2 +1,95 @@
...
+* Refer to the example below for how the structure of `./examples/<app_name>` directory would look like:
+
+```
+moveit_example_apps/examples/
```
Comment (davetcoleman) [inline README.md:32]: I think there's too much detail in this example app...

File: README.md [hunk 5]
```diff
@@ -1,2 +1,95 @@
...
+    |   |   ├── gripper/
+    |   |   ├── customizable_objects/  
+    |   |   └── meshes/
+    |   ├── CMakeLists.txt
+    |   ├── package.xml
+    |   └── README.md
+    └── <app_moveit_config_package>/
+```
+
+* The tutorials build with [rosdoc_lite](http://wiki.ros.org/rosdoc_lite) and [Sphinx](http://wiki.ros.org/Sphinx) builder. Each tutorial should live in its own subdirectory under the `./tutorials/doc/<app_name>` direcotry.
```
Comment (davetcoleman) [inline README.md:75]: directory

**Review by davetcoleman — COMMENTED** [review]
This PR is so big its causing my Google Chrome browser to crawl. Its taking 1.4 GB of my computer! Could you split this into two: the code and the tutorials, please?

**Comment by RoboticsYY** [comment]
> Could you split this into two: the code and the tutorials, please?

Yes, I can split the code and tutorials into two PRs.

File: README.md (hunk 1 above)
Comment (RoboticsYY) [reply README.md:15]: We can try to setup CI with moveit_ci. The difficulty lies in the dependencies of the app, that are compiled against caffe or OpenVINO, they all require multiple configurations.

File: README.md (hunk 2 above)
Comment (RoboticsYY) [reply README.md:27]: I will rectify all the `directory` in all the documents.

**Comment by RoboticsYY** [comment]
Split this PR into smaller ones, close this one.

**Pull request closed without merging**